File: dsa/binary-search/BinarySearch.java
public class BinarySearch {
    public static void main(String[] args) {
        int[] arr = { 10, 20, 30, 40, 50, 60, 70, 80, 90, 100 };
        int target = 30;
        int index = binarySearch(arr, target);
        System.out.println(index != -1 ? "Element found at index " + index : "Element not found");
    }

    private static int binarySearch(int[] arr, int target) {
        int min = 0;
        int max = arr.length - 1;
        while (min <= max) {
            int mid = min + (max - min) / 2;
            if (arr[mid] == target) {
                return mid;
            } else if (arr[mid] < target) {
                min = mid + 1;
            } else if (arr[mid] > target) {
                max = mid - 1;
            }
        }
        return -1;
    }
}

// https://www.khanacademy.org/computing/computer-science/algorithms/binary-search/a/binary-search

File: dsa/binary-search/binary-search.py
def binarySearch(array, x, low, high):
    while low <= high:
        mid = low + (high - low)//2
        if array[mid] == x:
            return mid
        elif array[mid] < x:
            low = mid + 1
        else:
            high = mid - 1
    return -1

array = input("Input  : ").split(" ")
target = input("Target : ")
result = binarySearch(array, target, 0, len(array)-1)
if result != -1:
    print("Output : " + str(result))
